op-bindings/bindgen/main.go

```diff
@@ -0,0 +1,250 @@
+			Name:     "metadata-out",
+			Usage:    "Output directory to put contract metadata files in",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "go-package",
+			Usage:    "Go package name given to generated files",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "monorepo-base",
+			Usage:    "Path to the base of the monorepo",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "log-level",
+			Usage:    "Configured the level of logs that will be printed, defaulted to info. Options: none, debug, info, warn, error, crit",
+			Required: false,
+		},
+	}
+}
+
+func localFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:     "local-contracts",
+			Usage:    "Path to file containing list of contracts to generate bindings for that have Forge artifacts available locally",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:  "source-maps-list",
+			Usage: "Comma-separated list of contracts to generate source-maps for",
+		},
+		&cli.StringFlag{
+			Name:     "forge-artifacts",
+			Usage:    "Path to forge-artifacts directory, containing compiled contract artifacts",
+			Required: true,
+		},
+	}
+}
+
+func remoteFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:     "client",
+			Usage:    "Name of remote client to connect to. Available clients: etherscan",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "remote-contracts",
+			Usage:    "Path to file containing list of contracts to generate bindings for that will have ABI and bytecode sourced from a remote source",
+			Required: true,
+		},
+		&cli.IntFlag{
+			Name:     "source-chainid",
+			Usage:    "Chain ID for the source chain contract data will be pulled from",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     "source-apikey",
+			Usage:    "API key to access remote source for source chain queries",
+			Required: true,
+		},
+		&cli.IntFlag{
+			Name:     "compare-chainid",
+			Usage:    "Chain ID for the chain contract data will be compared against",
+			Required: false,
+		},
+		&cli.StringFlag{
+			Name:     "compare-apikey",
+			Usage:    "API key to access remote source for contract data comparison queries",
+			Required: false,
+		},
+		&cli.IntFlag{
+			Name:  "api-max-retries",
+			Usage: "Max number of retries for getting a contract's ABI from Etherscan if rate limit is reached",
+			Value: 3,
+		},
+		&cli.IntFlag{
+			Name:  "api-retry-delay",
+			Usage: "Number of seconds before trying to fetch a contract's ABI from Etherscan if rate limit is reached",
+			Value: 2,
+		},
+		&cli.BoolFlag{
+			Name:  "compare-deployment-bytecode",
+			Usage: "When set to true, each contract's deployment bytecode retrieved from the source chain will be compared to the bytecode retrieved from the compare chain. If the bytecode differs between the source and compare chains, there's a possibility that the bytecode from the source chain may not work as intended on an OP chain",
+			Value: false,
```

Comment:
should these default be true? i feel like it should always be checked unless otherwise specified?

May not have enough context to know why false is a better default

Reply:
I think ideally yes, these check are always enabled, but because it requires a second API key network, and because it isn't strictly necessary for this CLI to function, I made it optional